drivers/spi/spi_esp32_spim.c

```diff
@@ -55,19 +58,43 @@ static inline void spi_esp32_complete(const struct device *dev,
 static int IRAM_ATTR spi_esp32_transfer(const struct device *dev)
 {
 	struct spi_esp32_data *data = dev->data;
+	const struct spi_esp32_config *cfg = dev->config;
 	struct spi_context *ctx = &data->ctx;
 	spi_hal_context_t *hal = &data->hal;
 	spi_hal_dev_config_t *hal_dev = &data->dev_config;
 	spi_hal_trans_config_t *hal_trans = &data->trans_config;
 	size_t chunk_len = spi_context_max_continuous_chunk(&data->ctx);
 	chunk_len = MIN(chunk_len, SOC_SPI_MAXIMUM_BUFFER_SIZE);
+	size_t bit_len = chunk_len << 3;
+	uint8_t *rx_temp = NULL;
+	uint8_t *tx_temp = NULL;
+
+	if (cfg->dma_enabled) {
+		bit_len = (!chunk_len) ? (chunk_len + 1) << 3 : bit_len;
+		if (ctx->tx_buf && !esp_ptr_dma_capable((uint32_t *)&ctx->tx_buf[0])) {
+			tx_temp = k_malloc(((ctx->tx_len << 3) + 7) / 8);
+			if (!tx_temp) {
+				LOG_ERR("Error allocating temp buffer Tx");
+				return -ENOMEM;
+			}
+			memcpy(tx_temp, &ctx->tx_buf[0], ((ctx->tx_len << 3) + 7) / 8);
```

Comment:
It would not be better to make this exotic length into a macro?

Reply:
Simplified the tx buffer length and added a comment on the rx buffer allocation :)